lib/build.gradle.kts

```
@@ -55,6 +69,104 @@ tasks.dokkaJavadoc {
+        return when (this) {
+            KOTLIN -> Regex("""```(${KOTLIN.langName}|kt)[\s\S]*?```""")
+            JAVA -> Regex("""```${JAVA.langName}[\s\S]*?```""")
+        }
+    }
+
+    fun getFunctionRegex(functionName: String): Regex {
+        return when (this) {
+            KOTLIN -> Regex("""fun\s+$functionName\s*\(.*\)\s*\{([\s\S]*?)\}""")
+            JAVA -> Regex(
+                """(?:public|private|protected)?\s+[\w<>\[\]]+\s+$functionName\s*\([^)]*\)\s*\{([\s\S]*?)}\s*""",
+                RegexOption.MULTILINE,
+            )
+        }
+
+    }
+}
+
+tasks.register("generateGuides") {
+    doLast {
+        val kotlinSamplesDir = "${project.projectDir}/src/test/kotlin"
+        val javaSamplesDir = "${project.rootDir}/javademo/src/main/java/com/example/javademo"
+        val guideFile = file("guide.md")
+        val kotlinGuideFile = project.projectDir.resolve("${Language.KOTLIN.langName}-guide.md")
+        val javaGuideFile = project.projectDir.resolve("${Language.JAVA.langName}-guide.md")
+
+        val guideContent = guideFile.readText()
+
+        val docsStartRegex = Regex("""^\s*//\s*docsStart\s*$""")
+        val docsEndRegex = Regex("""^\s*//\s*docsEnd\s*$""")
+
+        fun extractCodeSection(path: String, functionName: String, language: Language): String {
+            val file = file(path)
+            if (!file.exists()) {
+                return ""
+            }
+
+            val content = file.readText()
+
+            val matchResult = language.getFunctionRegex(functionName).find(content) ?: return ""
+            val codeSection = matchResult.groupValues[1].lines()
+            val filteredLines = mutableListOf<String>()
+            var capture = false
+
+            for (line in codeSection) {
+                when {
+                    docsStartRegex.containsMatchIn(line) -> capture = true
+                    docsEndRegex.containsMatchIn(line) -> capture = false
+                    capture -> filteredLines.add(line)
+                }
+            }
+            val minIndent = codeSection.filter { it.isNotBlank() }
+                .minOfOrNull { it.indexOfFirst { char -> !char.isWhitespace() } } ?: 0
+
+            return filteredLines.joinToString("\n") { if (it.isBlank()) it else it.drop(minIndent) }
+        }
+
+        fun processFileContent(dir: String, language: Language): String {
+            var content = language.getCodeSectionRegex().replace(guideContent) { matchResult ->
+                val path = matchResult.groupValues[1]
+                val functionName = matchResult.groupValues[2]
+                val codeSection = extractCodeSection(path="$dir/$path", functionName=functionName, language=language)
+                if (codeSection.isBlank()) "" else "```$language\n$codeSection\n```"
+            }
+
+            content = content.replace(Language.KOTLIN.getCodeSectionRegex(), "")
+                .replace(Language.JAVA.getCodeSectionRegex(), "")
```

Nit: 
To avoid duplication and simplify logic:

```suggestion
            content = Language.values().fold(content) { c, lang ->
                c.replace(lang.getCodeSectionRegex(), "")
            }
```